src/common/services/garApi.js

```diff
@@ -3,105 +3,104 @@ const logger = require('../utils/logger')(__filename);
 const endpoints = require('../config/endpoints');
 
 module.exports = {
+
   /**
-   * Updates GAR
+   * Updates GAR.
+   *
    * @param {String} garId id of GAR to be patched
    * @param {String} status new status of GAR
    * @param {Object} partial new GAR fields
    * @returns {Promise} resolves with API response
    */
+
   patch(garId, status, partial) {
     const reqBody = partial;
     reqBody.status = status;
-    return new Promise((resolve) => {
+    return new Promise((resolve, reject) => {
       request.patch({
         headers: { 'content-type': 'application/json' },
         url: endpoints.updateGar(garId),
         body: JSON.stringify(reqBody),
       }, (error, response, body) => {
+        logger.debug(`Patch GAR details returned status code: ${response && response.statusCode}`);
         if (error) {
           logger.error('Failed to call GAR put API endpoint');
-          return 'Failed to update GAR';
+          reject(error);
+          return;
         }
-        resolve(body);
         logger.debug('Successfully called GAR put endpoint');
-        return body;
-      });
-    })
-      .catch((err) => {
```

Comment:
Removed these catch blocks as they actually could hang. Same with adding in rejects. This is much more in line with Promises should have been done in the first place.

The hope is that more useful error messages will be logged.